### DISCUSSION TITLE
auth.uid() with custom backend

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#2379] [Questions] [answered]
How does this thing work **auth.uid()** if I implemented a custom backend for write operations?
In docs it says

1. A user signs up. Supabase creates a new user in the auth.users table.
2. Supabase returns a new JWT, which contains the user's UUID.
3. Every request to your database also sends the JWT.
4. Postgres inspects the JWT to determine the user making the request.
5. The user's UID can be used in policies to restrict access to rows.

My current process is I directly access the db connection string... How does policy **auth.uid()** knows who's ID?

**Comment by steve-chavez** [comment, marked as answer]
`auth.uid()` definition is:

```sql
create or replace function auth.uid() returns uuid as $$
  select nullif(current_setting('request.jwt.claim.sub', true), '')::uuid;
$$ language sql stable;
```

The `current_setting(request.jwt.claim.sub)` comes from PostgREST(see [here](https://postgrest.org/en/stable/api.html#accessing-request-headers-cookies-and-jwt-claims)), which decodes the JWT and sets its claims for PostgreSQL to consume.

**Reply by sandbox-apps** [reply]
So in short of saying I can't use the `auth.uid()` because I didn't connect using PostgREST. Because I directly used the database connection string

**Reply by sandbox-apps** [reply]
I should let you know that I'm using prisma for write operations (as it easy to code write operations in prisma) that's why I do direct connection with the postgres db. But wanting to still leverage the auth policies.

**Reply by sandbox-apps** [reply]
@steve-chavez can we also provide and use this if I use nestJS ?
`request.jwt.claim.sub`

**Reply by steve-chavez** [reply]
I guess you could decode the JWT yourself(in nestJS, don't know about it though) and then set it with `select set_config('request.jwt.claim.sub', 'any_value', true)` on each transaction - then the `current_setting` will work on the policies side.

**Reply by sandbox-apps** [reply]
I really dont get what you mean by this :(
`set it with select set_config('request.jwt.claim.sub', 'any_value', true) on each transaction`
